app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -306,70 +357,75 @@ def vms
         :snapshots        => snapshots(vm)
       )
 
-      vm_hardware(persister_vm, vm)
+      vm_hardware(persister_vm, vm, disks, template)
       operating_systems(persister_vm, vm)
       custom_attributes(persister_vm, vm)
     end
   end
 
   def storages(vm)
     storages = []
+    disks = []
     collector.collect_attached_disks(vm).to_miq_a.each do |disk|
+      next if disk.kind_of?(Array) && [].empty?
+
+      disks << disk
       disk.storage_domains.to_miq_a.each do |sd|
         storages << persister.storages.lazy_find(ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(sd.href))
       end
     end
     storages.compact!
     storages.uniq!
 
-    storages
+    return storages, disks
   end
 
-  def vm_hardware(presister_vm, vm)
+  def vm_hardware(presister_vm, vm, disks, template)
     topology = vm.cpu.topology
     cpu_socks = topology.try(:sockets) || 1
     cpu_cores = topology.try(:cores) || 1
 
     persister_hardware = persister.hardwares.find_or_build(presister_vm).assign_attributes(
       :guest_os             => vm.dig(:os, :type),
       :annotation           => vm.try(:description),
-      :cpu_cores_per_socket => cpu_socks,
-      :cpu_sockets          => cpu_cores,
+      :cpu_cores_per_socket => cpu_cores,
+      :cpu_sockets          => cpu_socks,
       :cpu_total_cores      => cpu_cores * cpu_socks,
       :memory_mb            => vm.memory / 1.megabyte
     )
 
-    hardware_disks(persister_hardware, vm)
-    addresses = hardware_networks(persister_hardware, vm)
+    hardware_disks(persister_hardware, disks)
+    addresses = hardware_networks(persister_hardware, vm, template)
     hardware_guest_devices(persister_hardware, vm, addresses)
   end
 
-  def hardware_networks(persister_hardware, vm)
+  def hardware_networks(persister_hardware, vm, template)
+    addresses = []
+    return addresses if template
+
     devices = collector.collect_vm_devices(vm)
-    device = devices[0]
+    devices = devices.reject(&:blank?) unless devices.nil?
+    return if devices.blank?
 
+    device = devices[0]
```

Comment:
if this is an array the next line will fail, maybe flatten should be used in line 407?